internal/platform/platform.go

```diff
@@ -0,0 +1,387 @@
+
+	var group string = "root/Level1A/ReconstructedData/PreciseOrbitEstimation"
+
+	time := getDatasetFloat64(filename, group, "Time")
+	gnssStateJ2000 := getDatasetFloat64(filename, group, "acsGnssStateJ2000")
+	uncertainty := getDatasetFloat64(filename, group, "acsNavigationUncertainty")
+
+	//uncertainty is common to all records
+	uncertainty2d := to6by6arr(uncertainty)
+
+	var records []OrbitRecord
+	n := len(time)
+	for i := range time {
+		record := OrbitRecord{
+			Time:        toDateTime(time[i]),
+			Uncertainty: uncertainty2d,
+		}
+		copy(record.GnssStateJ2000[:], to1DSlice(gnssStateJ2000, 6, n, i))
+		records = append(records, record)
+	}
+	return records
+}
+
+func getAttitudeRecords(filename string) []AttitudeRecord {
+
+	var group string = "root/Level1A/ReconstructedData/PreciseAttitudeEstimation"
+
+	time := getDatasetFloat64(filename, group, "Time")
+	attitudeState := getDatasetFloat64(filename, group, "afsAttitudeState")
+	attitudeUncertainty := getDatasetFloat64(filename, group, "afsAttitudeUncertainty")
+	rateUncertainty := getDatasetFloat64(filename, group, "afsRateUncertainty")
+	spacecraftRate := getDatasetFloat64(filename, group, "afsSpacecraftRate")
+	tangentPoint := getDatasetFloat64(filename, group, "afsTangentPoint")
+
+	var records []AttitudeRecord
+	n := len(time)
+	for i := range time {
+		record := AttitudeRecord{Time: toDateTime(time[i])}
+		copy(record.SpacecraftRate[:], to1DSlice(spacecraftRate, 3, n, i))
+		copy(record.TangentPoint[:], to1DSlice(tangentPoint, 3, n, i))
+		copy(record.AttitudeState[:], to1DSlice(attitudeState, 4, n, i))
+		record.AttitudeUncertainty = to3by3arr(attitudeUncertainty, i)
+		record.RateUncertainty = to3by3arr(rateUncertainty, i)
+		records = append(records, record)
+	}
+	return records
+}
+
+func to1DSlice(inslice []float64, sizeDim1 int, sizeDim2, offset int) []float64 {
+	//returns a 1-d slice extracted from a 1-d slice that is a representation
+	//of a 2-Darray with shape (sizeDim1, sizeDim2)
+	outslice := make([]float64, sizeDim1)
+	for i := 0; i < sizeDim1; i++ {
+		outslice[i] = inslice[offset+i*sizeDim2]
+	}
+	return outslice
+}
+
+func to3by3arr(inslice []float64, offset int) [3][3]float64 {
+	//returns a 3 x 3 array extracted from a 1-d slice that is a representation
+	//of an array with shape (len(inslice)/9, n, n)
+	var arr [3][3]float64
+	for i := 0; i < 3; i++ {
+		start := offset*9 + i*3
+		copy(arr[i][:], inslice[start:start+3])
+	}
+	return arr
+}
+
+func to6by6arr(inslice []float64) [6][6]float64 {
+	//returns a 6 x 6 array extracted from a 1-d slice of len 36
+	var arr [6][6]float64
+	for i := 0; i < 6; i++ {
+		copy(arr[i][:], inslice[i*6:i*6+6])
+	}
+	return arr
+}
+
+func toDateTime(secondsSinceEpoch float64) time.Time {
+	dt := time.Duration(secondsSinceEpoch*1e9) * time.Nanosecond
+	datetime := dateEpochGPS.Add(dt)
+	return datetime
+}
+
+//GetFilepath manipulates filename
+func GetFilepath(inputFile string, outputDirectory string) string {
```

Comment:
Please add tests for function.

How does it work if input doesn't end in `.something` for example.

Reply:
it is already tested. but the platform_test.go is so large so you have to load it below this file.

Reply:
I didn't find any unittests and it is an exported function so to me that means it ought to have its own testing.

Reply:
sure, but it has unit testing TestGetFilepath  in platform_test.go

Reply:
I was looking for it with search on page and didn't realize the entire file was hidden because the diff was so large. Sorry.